public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -506,34 +517,62 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
             subQueries.push(this.handleStreamingQuery(options, traceqlSearchTargets, queryFromFilters));
           } else {
             const startTime = performance.now();
+            const queries = traceqlSearchTargets.map((t) => ({ ...t, query: queryFromFilters }));
+
             subQueries.push(
-              this._request('/api/search', {
-                q: queryFromFilters,
-                limit: options.targets[0].limit ?? DEFAULT_LIMIT,
-                spss: options.targets[0].spss ?? DEFAULT_SPSS,
-                start: options.range.from.unix(),
-                end: options.range.to.unix(),
-              }).pipe(
-                map((response) => {
-                  reportTempoQueryMetrics('grafana_traces_traceql_response', options, {
-                    success: true,
-                    streaming: false,
-                    latencyMs: Math.round(performance.now() - startTime), // rounded to nearest millisecond
-                    query: queryFromFilters ?? '',
-                  });
-                  return {
-                    data: formatTraceQLResponse(
-                      response.data.traces,
-                      this.instanceSettings,
-                      targets.traceqlSearch[0].tableType
-                    ),
-                  };
+              super.query({ ...options, targets: queries }).pipe(
+                map((response: DataQueryResponse) => {
+                  if (queries[0].tableType === SearchTableType.Traces && response.data && response.data.length > 0) {
```

Comment:
Shouldn't we map queries/responses via refId? What's gonna happen if there are multiple queries with different types?

Reply:
this doesn't seem to be supported in the frontend implementation ([we just send the first query](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/tempo/datasource.ts#L510-L516)). and the goal of this is to replicate the frontend as closely as possible. i would love to support multiple queries but i think its out of scope of this pr.